### PR TITLE
[CAZ-1333] Redirect to compliant page when vehicle is exempted

### PR DESCRIPTION
**Pull request by musztardem**
## Motivation and Context

[Link to JIRA card](https://eaflood.atlassian.net/browse/CAZ-1333)

## Description

* Updated `VehicleDetails` model to include `exempt` field
* In `submit_details` action I added check which verifies if vehicle is exempted and redirects when needed.
* Added specs

## How Has This Been Tested?

* Manually
* With RSpec

## Screenshots (if appropriate):
Video is in the comment on JIRA card

## Checklist:

- [x] It contains only changes required by issue (does not contain other PR)
- [x] Includes link to a issue (if apply)
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: app/models/vehicle_details.rb [hunk 1]
```diff
@@ -67,6 +67,10 @@ def model
     string_field('model')
   end
 
+  def exempt?
```
Comment (miqs1992) [inline app/models/vehicle_details.rb]: Please add docs comment

File: app/controllers/vehicles_controller.rb [hunk 2]
```diff
@@ -52,6 +52,9 @@ def submit_details
     end
 
     store_vehicle_details
+    @vehicle_details = VehicleDetails.new(parsed_vrn)
```
Comment (miqs1992) [inline app/controllers/vehicles_controller.rb]: I think it is better to add this in `details` to avoid doubled calls - see VCCS for this

Comment (miqs1992) [reply app/controllers/vehicles_controller.rb]: https://github.com/InformedSolutions/JAQU-CAZ-Vehicle-Compliance-Checker-Web/blob/0dcdc64f23161891b5498a98f4742e723c0cc69e/app/controllers/vehicle_checkers_controller.rb#L79

File: app/controllers/vehicles_controller.rb [hunk 3]
```diff
@@ -67,6 +67,7 @@ def submit_details
   #
   def details
     @vehicle_details = VehicleDetails.new(vrn)
+    redirect_to(compliant_vehicles_path) && return if @vehicle_details.exempt?
```
Comment (likeuwill) [inline app/controllers/vehicles_controller.rb]: isn't better return `return redirect_to compliant_vehicles_path if @vehicle_details.exempt?` ?

File: spec/models/vehicle_details_spec.rb [hunk 4]
```diff
@@ -57,6 +58,12 @@
     end
   end
 
+  describe '.exempt?' do
```
Comment (likeuwill) [inline spec/models/vehicle_details_spec.rb]: from `vccs`, plz add same
```
  describe '.exempt?' do
    describe 'when key is not present' do
      it 'returns a nil' do
        expect(subject.exempt?).to eq(nil)
      end
    end

    describe 'when key is present' do
      before do
        allow(ComplianceCheckerApi).to receive(:vehicle_details).and_return('exempt' => true)
      end

      it 'returns a true' do
        expect(subject.exempt?).to eq(true)
      end
    end
  end
```
the problem is about nil value, if vehicle not exempted the api will not return `exempt` key in response (maybe now they will)

**Pull request merged**